### PR TITLE
feat: bundle with webpack

### PR DESCRIPTION
**Pull request by llorllale**
Bundle client and server versions of aries-js-worker with webpack.

TODO: publish the bundles

Signed-off-by: George Aristy <goerge.aristy@securekey.com>

### CODE REVIEW

File: cmd/aries-js-worker/package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  ],
+  "files": [
+    "src/aries.js",
+    "src/aries-worker.js",
+    "src/aries-worker-node.js",
+    "src/wasm_exec.js",
+    "dist/web/aries-js-worker.wasm"
+  ],
+  "author": "hyperledger/aries",
+  "license": "ISC",
```
Comment (troyronda) [inline cmd/aries-js-worker/package.json]: Apache-2.0

File: cmd/aries-js-worker/index.html [hunk 2]
```diff
@@ -11,11 +11,11 @@
 <html>
```
Comment (troyronda) [inline cmd/aries-js-worker/index.html]: should have license header comments.

File: cmd/aries-js-worker/App.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ SPDX-License-Identifier: Apache-2.0
 
 // This is a test nodejs app for developers to test WASM integration.
 
-const { Aries } = require('./aries.js')
+const { Aries } = require('./dist/node/aries.js')
```
Comment (troyronda) [inline cmd/aries-js-worker/App.js]: We should probably comment on the real NPM package path.

File: cmd/aries-js-worker/index.html [hunk 4]
```diff
@@ -11,11 +11,11 @@
...
 <head>
     <meta charset="utf-8"/>
-    <script src="aries.js"></script>
+    <script src="dist/web/aries.js"></script>
     <script>
         async function handleInput() {
             const invite = document.querySelector("#test-input").value
-            const connID = await Aries._test._echo(invite)
+            const connID = await Aries.Aries._test._echo(invite)
```
Comment (troyronda) [inline cmd/aries-js-worker/index.html]: TODO: revert this extra Aries.

File: cmd/aries-js-worker/package.json [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    "prepare": "npm run build",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+ssh://git@github.com/hyperledger/aries-framework-go.git"
+  },
+  "keywords": [
+    "aries",
+    "ssi",
```
Comment (troyronda) [inline cmd/aries-js-worker/package.json]: didcomm

Comment (troyronda) [inline cmd/aries-js-worker/package.json]: didcomm
hyperledger

File: cmd/aries-js-worker/package.json [hunk 6]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "@trustbloc/aries",
```
Comment (troyronda) [inline cmd/aries-js-worker/package.json]: shouldn’t be trustbloc

File: cmd/aries-js-worker/package.json [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  "version": "0.0.1",
+  "main": "dist/aries.js",
+  "module": "src/aries.js",
+  "description": "APIs for building SSI agents.",
```
Comment (troyronda) [inline cmd/aries-js-worker/package.json]: should be Aries agents

File: cmd/aries-js-worker/src/worker-impl-web.js [hunk 8]
```diff
@@ -4,7 +4,16 @@ Copyright SecureKey Technologies Inc. All Rights Reserved.
 SPDX-License-Identifier: Apache-2.0
 */
 
-self.importScripts("wasm_exec.js")
+const args = {};
+location.search.slice(1).split("&").forEach(param => {
```
Comment (troyronda) [inline cmd/aries-js-worker/src/worker-impl-web.js]: comment would be helpful.

File: cmd/aries-js-worker/src/worker-impl-web.js [hunk 9]
```diff
@@ -14,8 +23,9 @@ if (!WebAssembly.instantiateStreaming) { // polyfill
 }
 
 const go = new Go();
-WebAssembly.instantiateStreaming(fetch("aries-js-worker.wasm"), go.importObject).then(
-    (result) => { go.run(result.instance); }
+WebAssembly.instantiateStreaming(fetch(wasm), go.importObject).then(
+    result => { go.run(result.instance); },
+    err => { throw new Error("failed to fetch wasm blocb: " + err.message) }
```
Comment (troyronda) [inline cmd/aries-js-worker/src/worker-impl-web.js]: assume typo: blocb

File: cmd/aries-js-worker/package-lock.json [hunk 10]
```diff
@@ -0,0 +1,4457 @@
+{
+  "name": "@trustbloc/aries",
```
Comment (soluchok) [inline cmd/aries-js-worker/package-lock.json]: Do we need package-lock.json file?

File: cmd/aries-js-worker/index.html (hunk 2 above)
Comment (llorllale) [reply cmd/aries-js-worker/index.html]: I'm confused - it does have the license header comments lines 1->9.

File: cmd/aries-js-worker/src/worker-impl-web.js (hunk 9 above)
Comment (llorllale) [reply cmd/aries-js-worker/src/worker-impl-web.js]: done

File: cmd/aries-js-worker/src/worker-impl-web.js (hunk 8 above)
Comment (llorllale) [reply cmd/aries-js-worker/src/worker-impl-web.js]: done

File: cmd/aries-js-worker/package.json (hunk 7 above)
Comment (llorllale) [reply cmd/aries-js-worker/package.json]: done

File: cmd/aries-js-worker/package.json (hunk 6 above)
Comment (llorllale) [reply cmd/aries-js-worker/package.json]: done

File: cmd/aries-js-worker/package.json (hunk 5 above)
Comment (llorllale) [reply cmd/aries-js-worker/package.json]: done

File: cmd/aries-js-worker/index.html (hunk 4 above)
Comment (llorllale) [reply cmd/aries-js-worker/index.html]: done

File: cmd/aries-js-worker/package.json (hunk 1 above)
Comment (llorllale) [reply cmd/aries-js-worker/package.json]: done

File: cmd/aries-js-worker/package-lock.json (hunk 10 above)
Comment (llorllale) [reply cmd/aries-js-worker/package-lock.json]: I think the answer would be yes: consistency in the development workflow and consistency in the CI.

Comment (troyronda) [reply cmd/aries-js-worker/package-lock.json]: @llorllale where did we land on this topic?

Comment (llorllale) [reply cmd/aries-js-worker/package-lock.json]: Yes, reproducible builds are good. For instance, notice that not all of the transitive dependency versions are locked down.

**Review by troyronda — APPROVED** [review]

**Review by fqutishat — APPROVED** [review]

**Pull request merged**